src/v/utils/stream_utils.h

```diff
@@ -138,6 +138,19 @@ class input_stream_fanout final {
           _num_clients);
         unsigned mask = 1U << index;
         std::optional<ss::temporary_buffer<Ch>> buf = std::nullopt;
+
+        auto all_bits = (1U << _num_clients) - 1;
```

Comment:
This logic could be moved to the `detach` method. Right after the bit that corresponds to detached client is set you need to do the same thing. This code is in the fast path and executed for every read operation. But `detach` is only performed when the client detaches itself from the fanout.

Reply:
sure, moved to detach method